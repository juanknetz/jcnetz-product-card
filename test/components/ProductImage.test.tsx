import React from 'react';
import renderer from 'react-test-renderer';
import { ProductImage } from '../../src/components';
import { product2 } from '../data/products';

describe('ProductImage', () => {
  test('debe mostrar el componente correctamente con la imagen personalizada', () => {
    const wrapper = renderer.create(
      <ProductImage img="https://image.png" className="custom-class" />
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('debe mostrar el componente correctamente CON la imagen', () => {
    const wrapper = renderer.create(
      <ProductImage img={product2.img} className="custom-class" />
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
