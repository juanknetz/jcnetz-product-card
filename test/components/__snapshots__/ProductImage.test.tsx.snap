// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage debe mostrar el componente correctamente CON la imagen 1`] = `
<img
  alt="Product"
  className="productImg custom-class"
  src="./coffee-mug.png"
/>
`;

exports[`ProductImage debe mostrar el componente correctamente con la imagen personalizada 1`] = `
<img
  alt="Product"
  className="productImg custom-class"
  src="https://image.png"
/>
`;
